wasm-syntax.md

```diff
@@ -28,25 +28,27 @@ $2^32$ and $2^64$ are used often enough to warrant providing macros for them.
     rule #pow64 => 18446744073709551616 [macro]
 ```
 
-### Values
+### Types
 
-WASM values are either integers or floating-point numbers, of 32- or 64-bit widths.
-First, the type of values is built, then values annotated with their types are provided.
+WASM has four basic types, for 32 and 64 bit integers and floats.
+There are two basic type-constructors: sequencing (`[_]`) and function spaces (`_->_`).
 
 ```k
-    syntax Number ::= Int | Float
- // -----------------------------
-
     syntax IValType ::= "i32" | "i64"
     syntax FValType ::= "f32" | "f64"
     syntax  ValType ::= IValType | FValType
  // ---------------------------------------
 
-    syntax IVal ::= "<" IValType ">" Int
-    syntax FVal ::= "<" FValType ">" Float
-    syntax  Val ::= "<"  ValType ">" Number
-                  | IVal | FVal
- // ---------------------------
+    syntax ValTypes ::= List{ValType, ""}
+    syntax ValType  ::= "[" ValTypes "]"
```

Comment:
It is a bit confusing that this allows nested types: `[ [ i32 [ i32 ] [ i32 i32 i32 ] ] i32 ]`

Is there a case where this is needed? If not, perhaps restricting this to a single vector is nicer.

Reply:
I have adjusted the syntax to only allow one level of nesting, which is tighter to the WASM spec.